### PR TITLE
docs: add `_secret_questions` setting

### LINKED ISSUE #809: `_secret_questions` setting is undocumented
The `_secret_questions` setting

https://github.com/copier-org/copier/blob/608654a03e748a3ed1b032c089e20b2811a0cfd3/tests/test_answersfile.py#L34-L35

is not documented. I think it's a redundant setting since a question supports the `secret` flag as well. In any case, the docs and capabilities should be in sync.

Which solution do you prefer?

1. Document this setting.
2. Remove support for this setting in the code.

### PR DESCRIPTION
**Pull request by sisp**
I've added documentation for the `_secret_questions` setting.

Closes #809.

### CODE REVIEW

File: docs/configuring.md [hunk 1]
```diff
@@ -830,6 +830,26 @@ Suppress status output.
 
     Not supported in `copier.yml`.
 
+### `secret_questions`
+
+-   Format: `List[str]`
+-   CLI flags: N/A
+-   Default value: N/A
```
Comment (yajo) [inline docs/configuring.md:837]:
```suggestion
-   Default value: `[]`
```

**Review by yajo — CHANGES_REQUESTED** [review]
I think you need to change this little thing.

File: docs/configuring.md (hunk 1 above)
Comment (sisp) [reply docs/configuring.md:837]: Sounds good.

But then I think [`migrations`](https://copier.readthedocs.io/en/latest/configuring/#migrations), [`skip_if_exists`](https://copier.readthedocs.io/en/latest/configuring/#skip_if_exists), and [`tasks`](https://copier.readthedocs.io/en/latest/configuring/#tasks) also need an empty list as the default value. By the way, [`jinja_extensions`](https://copier.readthedocs.io/en/latest/configuring/#jinja_extensions) already has an empty list as the default value.

I can create a separate PR for the other corrections, if you agree that's how it should be.

Comment (sisp) [reply docs/configuring.md]: WDYT about the other settings, @yajo?

Comment (yajo) [reply docs/configuring.md]: Yes, we should correct them too 👍

**Pull request merged**